src/gui/buttons.py

```diff
@@ -62,8 +118,64 @@ def get_new_missions_handler(event: UIEvent) -> UIEventHandler:
     eng.refresh_mission_board()
 
 
-def get_new_missions_button() -> UIFlatButton:
-    btn = UIFlatButton(text="New Missions")
+def get_new_missions_button() -> UITextureButton:
+    main_tex = PixelatedNinePatch(
+        left=1, right=1, bottom=3, top=1, texture=WindowData.buttons[7]
+    )
+    pressed_tex = PixelatedNinePatch(
+        left=1, right=1, bottom=3, top=3, texture=WindowData.buttons[9]
+    )
+    hovered_tex = PixelatedNinePatch(
+        left=1, right=1, bottom=3, top=1, texture=WindowData.buttons[11]
+    )
+    btn = UITextureButton(
+        text="New Missions",
+        texture=main_tex,
+        texture_hovered=hovered_tex,
+        texture_pressed=pressed_tex,
+    )
     btn.on_click = get_new_missions_handler
 
     return btn
+
+
+def end_turn_handler(view) -> UIEventHandler:
+    def _handle(event: UIEvent):
+        if not view.target_selection and eng.awaiting_input:
+            eng.next_combat_action()
+            eng.awaiting_input = False
+            return
+
+        if not view.target_selection:
+            return
+
+        ok = view.target_selection.confirm()
+        if ok:
+            view.combat_grid_section.hide_path()
+            view.target_selection = None
+            view.item_menu_mode_allowed = True
+
+    return _handle
+
+
+def end_turn_button(view) -> UITextureButton:
+    main_tex = PixelatedNinePatch(
+        left=1, right=1, bottom=3, top=1, texture=WindowData.buttons[7]
+    )
+    pressed_tex = PixelatedNinePatch(
+        left=1, right=1, bottom=3, top=3, texture=WindowData.buttons[9]
+    )
+    hovered_tex = PixelatedNinePatch(
+        left=1, right=1, bottom=3, top=1, texture=WindowData.buttons[11]
+    )
+
+    btn = UITextureButton(
+        text="CLICK ME TO ADVANCE!",
+        texture=main_tex,
+        texture_pressed=pressed_tex,
+        texture_hovered=hovered_tex,
+    )
+
+    btn.on_click = end_turn_handler(view)
+
+    return btn

```

Comment:
```suggestion

button_textures = {
    "texture": PixelatedNinePatch(
        left=1, right=1, bottom=3, top=1, texture=WindowData.buttons[7]
    )
    "texture_pressed": PixelatedNinePatch(
        left=1, right=1, bottom=3, top=3, texture=WindowData.buttons[9]
    )
    "texture_hovered": PixelatedNinePatch(
        left=1, right=1, bottom=3, top=1, texture=WindowData.buttons[11]
    )
}


def end_turn_button(view) -> UITextureButton:
    btn = UITextureButton(
        text="CLICK ME TO ADVANCE!",
        **button_textures
    )

    btn.on_click = end_turn_handler(view)

    return btn

```

Reply:
^ This will allow you to share texture configuration amongst your button factory functions